providers/flagd/README.md

```diff
@@ -15,6 +15,7 @@ You may also provide additional options to configure the provider client
 flagd.WithService(HTTP | HTTPS | GRPC)  // defaults to http 
 flagd.WithHost(string)                  // defaults to localhost
 flagd.WithPort(uint16)                  // defaults to 8013
+flagd.FromEnv()                         // sets the provider configuration from environemnt variables
```

Comment:
I agree on this.

A solution could be to track that a variably has been explicitly set and not update it in the `FromEnv` call in that case.

Reply:
I've taken a similar approach to this, but set the default values after the options have been run (if a value had been explicitly set to its default value it would have been overwritten)
Ive added a comment to explain the `-1` value for the ServiceProvider is to avoid the use of `0` which would evaluate to `HTTP`

Reply:
Nice. Can a test be added to demonstrate that this works?

Reply:
will do

Reply:
I'm still a bit confusing about this. Could we chat about it today?